app/support/payoff_manager.rb

```diff
@@ -0,0 +1,103 @@
+## Loan Manager to determine funding channel status
+## Determines using current month history and construction activity
+require 'byebug'
+module PayoffManager
+  include Days360
+  DISCHARGE_FEE = 250
+
+  def create_payoff(date_string)
+    date = Date.parse(date_string)
+    @payoff = payoffs.build(
+      payoff_date: date
+    )
+    create_loan_amount_line_item
+    create_unused_rehab_budget_line_item
+    create_interest_line_items(date)
+    create_closing_fee_line_items
+    @payoff.save!
+    @payoff
+  end
+
+  private
+
+  def create_unused_rehab_budget_line_item
+    unused_rehab = -(rehab_budget_amount -
+      processed_construction_draws.sum(:amount))
+    line_item = @payoff.line_items.build(
+      amount: unused_rehab,
+      item_type: LineItem.item_types[:unused_rehab_funds],
+      status: LineItem.statuses[:closed]
+    )
+    line_item.save!
```

Comment:
same thing with this pattern.  do we have to save this here or can we change state here and save in the controller?

Reply:
Okay, so I think a better way to address this is to only hit the payoffController once.
IN the payoff controller I can use `accepts_nested_attributes_for :line_items` to create the payoff AND all the line items at one time. Then I avoid hitting the db for each and every line item.